### PR TITLE
[google_maps_flutter] Annotate deprecated member usage

### PR DESCRIPTION
**Pull request by stuartmorgan**
There are several intentional uses of deprecated members from other packages within the plugin group, so annotate them so that they don't show up in the regular team audits:
- App-facing package integration tests of the deprecated style method.
- Handling of deprecated marker bitmap formats.

These turned up in the [routine repo audit](https://github.com/flutter/flutter/blob/main/docs/infra/Packages-Gardener-Rotation.md#deprecations).

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart [hunk 1]
```diff
@@ -194,9 +194,10 @@ class GoogleMapController {
   /// own configuration and are rendered on top of a GMap instance later. This
   /// happens in the second half of this method.
   ///
-  /// This method is eagerly called from the [GoogleMapsPlugin.buildView] method
-  /// so the internal [GoogleMapsController] of a Web Map initializes as soon as
-  /// possible. Check [_attachMapEvents] to see how this controller notifies the
+  /// This method is eagerly called from the
+  /// [GoogleMapsPlugin.buildViewWithConfiguration] method so the internal
```
Comment (stuartmorgan) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart:198]: The implementation had already changed to `buildViewWithConfiguration`, this comment was just outdated.

**Comment by stuartmorgan** [comment]
CHANGELOG/version override: The trivial doc comment change isn't worth publishing, especially since it's not API clients would be expected to use.

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart (hunk 1 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/google_maps_controller.dart:198]: Good catch!

**Review by ditman — APPROVED** [review]
maps and maps_web LGTM!

**Pull request merged**